code/game/objects/items/beacon.dm

```diff
@@ -1,49 +1,58 @@
-/obj/item/device/radio/beacon
-	name = "Tracking Beacon"
-	desc = "A beacon used by a teleporter."
-	icon_state = "beacon"
-	item_state = "signaler"
-	var/code = "electronic"
-	origin_tech = "bluespace=1"
-
-/obj/item/device/radio/beacon/Hear(message, atom/movable/speaker, var/datum/language/speaking, raw_message, radio_freq)
-	return
-
-
-/obj/item/device/radio/beacon/send_hear()
-	return null
-
-
-/obj/item/device/radio/beacon/verb/alter_signal(t as text)
-	set name = "Alter Beacon's Signal"
-	set category = "Object"
-	set src in usr
-
-	if ((usr.canmove && !( usr.restrained() )))
-		src.code = t
-	if (!( src.code ))
-		src.code = "beacon"
-	src.add_fingerprint(usr)
-	return
-
-
-/obj/item/device/radio/beacon/bacon //Probably a better way of doing this, I'm lazy.
-	proc/digest_delay()
-		spawn(600)
-			del(src)
-
-
-// SINGULO BEACON SPAWNER
-
-/obj/item/device/radio/beacon/syndicate
-	name = "suspicious beacon"
-	desc = "A label on it reads: <i>Activate to have a singularity beacon teleported to your location</i>."
-	origin_tech = "bluespace=1;syndicate=7"
-
-/obj/item/device/radio/beacon/syndicate/attack_self(mob/user as mob)
-	if(user)
-		user << "<span class='notice'>Locked In</span>"
-		new /obj/machinery/singularity_beacon/syndicate( user.loc )
-		playsound(src, 'sound/effects/pop.ogg', 100, 1, 1)
-		del(src)
-	return
\ No newline at end of file
+var/global/list/obj/item/beacon/beacons = list()
+
+/obj/item/beacon
+	name = "Tracking Beacon"
+	desc = "A beacon used by a teleporter."
+	icon_state = "beacon"
+	item_state = "signaler"
+	var/code = "electronic"
+	var/frequency = 1459
+	origin_tech = "bluespace=1"
+	flags = FPRINT
+
+/obj/item/beacon/New()
+	..()
+	beacons += src
+
+/obj/item/beacon/Destroy()
+	..()
+	beacons -= src
+
+/obj/item/beacon/attack_self(mob/user as mob)
+	..()
+	var/newfreq = input(user, "Input a new frequency for the beacon", "Frequency", null) as num
+	if(!newfreq)
+		return
+	frequency = sanitize_frequency(format_frequency(newfreq))
+
+/obj/item/beacon/verb/alter_signal(t as text)
+	set name = "Alter Beacon's Signal"
+	set category = "Object"
+	set src in usr
+
+	if ((usr.canmove && !( usr.restrained() )))
+		src.code = t
+	if (!( src.code ))
+		src.code = "beacon"
+	src.add_fingerprint(usr)
+	return
+
+
+/obj/item/beacon/bacon/proc/digest_delay()
```

Comment:
I uh... I don't actually know. It was there before but it's not as if its used anywhere.